### ISSUE TITLE
MPI_Type_get_extent incorrect for empty datatypes on linux

### ISSUE THREAD

**Issue by simonbyrne** [open-mpi/ompi#7266]
### What version of Open MPI are you using? (e.g., v1.10.3, v2.1.0, git branch name and hash, etc.)

4.0.1 and 3.1.4

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

Using cluster-provided version

### Please describe the system on which you are running

* Operating system/version: linux
* Computer hardware: x86_64
* Network type: infiniband

-----------------------------

## Details of the problem

When constructing an empty datatype, `MPI_Type_get_extent` gives invalid values, e.g.

```C
#include <mpi.h>
#include <stdio.h>

int main(int argc, char** argv) {
    MPI_Init(NULL, NULL);

    MPI_Datatype EmptyType;
    MPI_Type_contiguous(0, MPI_INT, &EmptyType);

    MPI_Aint lb, extent;
    MPI_Type_get_extent(EmptyType, &lb, &extent);

    printf("lowerbound = %ld\nextent = %ld\n", (long)lb, (long)extent);

    MPI_Finalize();
}
```
On Linux will print
```
lowerbound = 9223372036854775807
extent = 1
```

On MacOS, it prints
```
lowerbound = 0
extent = 0
```

**Comment by bosilca** [comment]
#7267 should fix the problem on the 3 stable. It appears to be already applied to the 4 thou.

**Comment by simonbyrne** [comment]
Ah, indeed this does appear to be a duplicate of #6575, and looks to be fixed by #6647.
